zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -366,6 +376,30 @@ public void addConfigNode() {
         }
     }
 
+    /**
+     * create a /zookeeper/uri-domain-map node for maintaining client-URIs to domain mapping for
+     * zookeeper
+     */
+    public void addUriDomainMapNode() {
+        if (!X509AuthenticationConfig.getInstance().isX509ZnodeGroupAclEnabled()) {
+            return;
+        }
+        DataNode zookeeperZnode = nodes.get(procZookeeper);
+        if (zookeeperZnode != null) { // should always be the case
+            zookeeperZnode.addChild(uriDomainMapChildZookeeper);
+        } else {
+            assert false : "There's no /zookeeper znode - this should never happen.";
+        }
+
+        nodes.put(uriDomainMapZookeeper, new DataNode(new byte[0], -1L, new StatPersisted()));
+        try {
+            // Reconfig node is access controlled by default (ZOOKEEPER-2014).
+            setACL(uriDomainMapZookeeper, ZooDefs.Ids.OPEN_ACL_UNSAFE, -1);
+        } catch (KeeperException.NoNodeException e) {
+            assert false : "There's no " + uriDomainMapZookeeper + " znode - this should never happen.";
```

Comment:
Is this check necessary? Consider just wrapping the whole block in a `try-catch` with a proper exception log?

Reply:
Same as above comment